client/client.go

```diff
@@ -0,0 +1,766 @@
+package client
```

Comment:
@coderabbitai read client/client.go and generate API documentation in Markdown.

Reply:
@coderabbitai read client/client.go and generate usage and examples documentation in Markdown format.